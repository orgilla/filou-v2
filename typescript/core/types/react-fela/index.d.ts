declare module "react-fela" {
  export const FelaComponent: any;
  export const FelaTheme: any;
}
