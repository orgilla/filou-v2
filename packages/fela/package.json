{
  "version": "0.14.8",
  "name": "@orgilla/fela",
  "description": "",
  "main": "lib",
  "modules": "es",
  "author": "Orgilla <hi@orgil.la>",
  "contributors": [
    "Benjamin Kniffler <bkniffler@me.com>",
    "Philipp Achenbach <olymp@philipp-achenbach.de>"
  ],
  "license": "MIT",
  "dependencies": {
    "fela": "^6.2.1",
    "fela-plugin-custom-property": "^7.0.3",
    "fela-plugin-embedded": "^5.2.11",
    "fela-plugin-extend": "^6.0.9",
    "fela-plugin-fallback-value": "^5.0.18",
    "fela-plugin-friendly-pseudo-class": "^5.0.17",
    "fela-plugin-named-keys": "^1.0.3",
    "fela-plugin-prefixer": "^5.0.19",
    "fela-plugin-unit": "^5.0.17",
    "tinycolor2": "^1.4.1"
  },
  "peerDependencies": {
    "react": "^16.4.2",
    "react-dom": "^16.4.2",
    "react-fela": "^8.0.0"
  }
}
