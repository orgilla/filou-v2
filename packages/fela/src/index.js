export { default as withTheme } from './with-theme';
export { default as withPulse } from './with-pulse';
export { default as colorsMaterial } from './colors-material';
export { default as ColorsProvider, getColor } from './colors-provider';

export { createComponent, Provider } from 'react-fela';
export { default as color } from 'tinycolor2';
export { default as tinyColor } from 'tinycolor2';

export { default as createRenderer } from './create-renderer';
export { default as ThemeProvider } from './theme-provider';
export { default as normalize } from './normalize';
export { default as withStyle } from './with-style';
