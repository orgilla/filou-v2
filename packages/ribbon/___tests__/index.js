import 'jest';
import React from 'react';
import renderer from 'react-test-renderer';
import { createRenderer } from 'fela';
import { Provider as FelaProvider, ThemeProvider } from 'react-fela';
import { renderToMarkup } from 'fela-dom';
import Ribbon from '../src';

const getProvider = (fela, theme = {}) => {
  const Provider = ({ children }) => (
    <FelaProvider rehydrate renderer={fela}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </FelaProvider>
  );
  return Provider;
};

describe('Ribbon', () => {
  it('should render correctly', () => {
    const fela = createRenderer();
    const Provider = getProvider(fela);
    const component = renderer.create(
      <Provider>
        <Ribbon>
          <Ribbon.Title brand="Heya" />
          Text
        </Ribbon>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(renderToMarkup(fela)).toMatchSnapshot();
  });
});
