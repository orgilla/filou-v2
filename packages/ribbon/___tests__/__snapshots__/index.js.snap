// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ribbon should render correctly 1`] = `
<div
  className="a b c d e"
>
  <nav
    className="a f g h i j k l m n o p q r s t u v w x y z ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw"
  >
    <span
      className="brand"
    >
      Heya
    </span>
    <div
      className="ax ay a"
    >
      <button
        aria-label="minimize"
        className="az ba bb bc bd be bf bg bh bi bj bk bl bm bn bo bp"
        onClick={[Function]}
        tabIndex="-1"
      >
        <svg
          aria-hidden="true"
          height="10"
          version="1.1"
          width="10"
        >
          <path
            d="M 0,5 10,5 10,6 0,6 Z"
          />
        </svg>
      </button>
      <button
        aria-label="maximize"
        className="az ba bb bc bd be bf bg bh bi bj bk bl bm bn bo bp"
        onClick={[Function]}
        tabIndex="-1"
      >
        <svg
          aria-hidden="true"
          height="10"
          version="1.1"
          width="10"
        >
          <path
            d="M 0,0 0,10 10,10 10,0 Z M 1,1 9,1 9,9 1,9 Z"
          />
        </svg>
      </button>
      <button
        aria-label="close"
        className="az ba bb bc bd be bf bg bh bi bj bk bl bm bn bo bp"
        onClick={[Function]}
        tabIndex="-1"
      >
        <svg
          aria-hidden="true"
          height="10"
          version="1.1"
          width="10"
        >
          <path
            d="M 0,0 0,0.7 4.3,5 0,9.3 0,10 0.7,10 5,5.7 9.3,10 10,10 10,9.3 5.7,5 10,0.7 10,0 9.3,0 5,4.3 0.7,0 Z"
          />
        </svg>
      </button>
    </div>
  </nav>
  Text
</div>
`;

exports[`Ribbon should render correctly 2`] = `"<style type=\\"text/css\\" data-fela-rehydration=\\"67\\" data-fela-type=\\"RULE\\">.a{display:flex}.b{width:100vw}.c{height:100vh}.d{flex:1}.e{flex-direction:column}.f{flex-direction:row}.g{padding-left:8}.h{min-height:32}.i{max-height:32}.j{padding-right:0}.k{align-content:center}.l{align-items:stretch}.m> .brand:first-child{position:relative}.n> .brand{transition:.3s cubic-bezier(.25,.8,.5,1),color 1ms}.o{-webkit-user-select:none}.p{-webkit-app-region:drag}.q .logo{padding-y:3}.r> a{-webkit-app-region:no-drag}.s> a{padding-x:6}.t> a{min-width:22}.u> a{display:flex}.v> a{height:initial}.w> a{align-items:center}.x> a{font-size:14}.y> a{justify-content:center}.z> a{margin-x:0}.ab> a{margin-y:0}.ac> a{outline:0}.ae> a{text-decoration:none}.af> a{position:relative}.ag> a{vertical-align:middle}.ah> a{user-select:none}.ai> span{-webkit-app-region:no-drag}.aj> span{padding-x:6}.ak> span{min-width:22}.al> span{display:flex}.am> span{height:initial}.an> span{align-items:center}.ao> span{font-size:14}.ap> span{justify-content:center}.aq> span{margin-x:0}.ar> span{margin-y:0}.as> span{outline:0}.at> span{text-decoration:none}.au> span{position:relative}.av> span{vertical-align:middle}.aw> span{user-select:none}.ax{flex-grow:0}.ay{flex-shrink:0}.az> svg{fill:#fff}.ba{-webkit-app-region:no-drag}.bb{display:inline-block}.bc{position:relative}.bd{width:45px}.be{box-shadow:none}.bf{color:rgb(160, 160, 160)}.bg{background-color:transparent}.bh{padding:0px}.bi{margin:0px}.bj{overflow:hidden}.bk{border-width:initial}.bl{border-style:none}.bm{border-color:initial}.bn{border-image:initial}.bo{border-radius:0px}.bp{transition:background-color 0.25s ease}</style>"`;
